### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by wendigo**
I want to do it early in the release cycle to detect regressions as fast as possible

The only functional change is that BigQuery arrow implementation no longer requires `--add-opens` JVM flags, which is reflected in the documentation.

### CODE REVIEW

**Review by martint — APPROVED** [review]

File: plugin/trino-bigquery/pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
             <dependency>
                 <groupId>com.google.cloud</groupId>
                 <artifactId>libraries-bom</artifactId>
-                <version>26.14.0</version>
+                <version>26.18.0</version>
```
Comment (ebyhr) [inline plugin/trino-bigquery/pom.xml:25]: > Update apache arrow to 12.0.1

The commit title and the actual change don't match.

Comment (wendigo) [reply plugin/trino-bigquery/pom.xml:25]: Yeah. I started with the arrow but in order to update it i needed to sync other dependencies like bigquery or protobuf. Logically it’s a single unit of change. I’ll improve commit message

Comment (wendigo) [reply plugin/trino-bigquery/pom.xml:25]: I squashed two commits together and fixed H2 test assertion which was the only failure in the tests.

**Review by electrum — APPROVED** [review]

**Comment by wendigo** [comment]
/test-with-secrets sha=b09f60299320edca03a618495dd1abad4b9c5d4a

**Comment by wendigo** [comment]
Unfortunately Arrow does not work without add-opens even for 12.0.1 :(

**Comment by wendigo** [comment]
/test-with-secrets sha=835f4d4774db91653340b3d6d7144fe4a1ff6511

**Pull request merged**